common/chains/client/types/node.go

```diff
@@ -0,0 +1,57 @@
+package types
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type NodeClientAPI[
+	CHAIN_ID types.ID,
+	BLOCK_HASH types.Hashable,
+	HEAD types.Head[BLOCK_HASH],
+	SUB types.Subscription,
```

Comment:
I think this can be removed. The only usage is the `Subscribe` method return arg, and the only usage of those values are direct access to the `types.Subscription` methods, so that type can be used directly.
```suggestion
```

Reply:
I was a bit reluctant to change it because it would mean the chain-specific Rpc_client would have to look like this:
`func (r *rpcClient) Subscribe(ctx context.Context, channel chan<- *evmtypes.Head, args ...interface{}) (commontypes.Subscription, error) {`
instead of this:
`func (r *rpcClient) Subscribe(ctx context.Context, channel chan<- *evmtypes.Head, args ...interface{}) (ethereum.Subscription, error) {`
but it does help declutter the code by removing the SUB type across components, so I'll give it a go and see how it looks.